modules/@apostrophecms/task/index.js

```diff
@@ -15,62 +15,90 @@
 
 const _ = require('lodash');
 const { stripIndent } = require('common-tags');
+const { SemanticAttributes } = require('@opentelemetry/semantic-conventions');
 
 module.exports = {
   options: { alias: 'task' },
   handlers(self) {
     return {
       'apostrophe:run': {
-        async runTask(isTask) {
+        async runTask(isTask, parentSpan) {
 
           if (!isTask) {
             return;
           }
 
-          let task;
           const cmd = self.apos.argv._[0];
-          if (!cmd) {
-            throw new Error('There is no command line argument to serve as a task name, should never happen');
-          }
-
-          if (cmd === 'help') {
+          const telemetry = self.apos.telemetry;
+          const spanName = `task:${cmd}`;
+          await telemetry.aposStartActiveSpan(spanName, async (span) => {
+            span.setAttribute(SemanticAttributes.CODE_FUNCTION, 'runTask');
```

Comment:
The attributes seem a little redundant in combination with the span name, but if they improve the presentation it's worth it.

Reply:
Attributes are more important than span names (called `operations` in the most back-ends). They are used to do the actual data slices, while names are the context and are representing the relationships.

You can think of a span name as "mongodb collection" and "attributes" as "the mongodb criteria", but there is more to that. 
Span name is the graph node name, and you could filter everything related to this node via attributes query and visualize it in all possible ways (table, grid, graph, sky is the limit).

Reply:
Example (Jaeger back-end):  
Operation: all  
Tags: `code.function="toArray" code.namespace="@apostrophecms/redirect"`  
Results: All GET/POST/etc root traces having `@apostrophecms/redirect:toArray` invocation.